src/observers/slack_resource.cpp

```diff
@@ -38,6 +38,16 @@ Try<Nothing> SlackResourceObserver::consume(const ResourceUsage& usage) {
     }
   }
 
+  Resources totalAgentResources(usage.total());
+  Option<double_t> totalAgentCpus = totalAgentResources.cpus();
+
+  if (totalAgentCpus.isSome()) {
+    double_t maxSlack = maxOversubscriptionFraction * totalAgentCpus.get();
```

Comment:
These calculation does not fix all cases:

IMO:
`MaxSlack = FRACTION * totalCpus - usedCpus;`